hyperspy/io_plugins/zspy.py

```diff
@@ -0,0 +1,575 @@
+                        zarrgroup2dict(
+                            group[key], lazy=lazy).items()
+                    ))])
+            elif key.startswith('_list_'):
+                dictionary[key[7 + key[6:].find('_'):]] = \
+                    [i for k, i in sorted(iter(
+                        zarrgroup2dict(
+                            group[key], lazy=lazy).items()
+                    ))]
+            elif key.startswith('_tuple_'):
+                dictionary[key[8 + key[7:].find('_'):]] = tuple(
+                    [i for k, i in sorted(iter(
+                        zarrgroup2dict(
+                            group[key], lazy=lazy).items()
+                    ))])
+            else:
+                dictionary[key] = {}
+                zarrgroup2dict(
+                    group[key],
+                    dictionary[key],
+                    lazy=lazy)
+    return dictionary
+
+
+def write_signal(signal, group, f=None,  **kwds):
+    """Writes a hyperspy signal to a zarr group"""
+
+    group.attrs.update(get_object_package_info(signal))
+    metadata = "metadata"
+    original_metadata = "original_metadata"
+
+    if 'compressor' not in kwds:
+        kwds['compressor'] = None
+
+    for axis in signal.axes_manager._axes:
+        axis_dict = axis.get_axis_dictionary()
+        coord_group = group.create_group(
+            'axis-%s' % axis.index_in_array)
+        dict2zarrgroup(axis_dict, coord_group, **kwds)
+    mapped_par = group.create_group(metadata)
+    metadata_dict = signal.metadata.as_dictionary()
+    overwrite_dataset(group, signal.data, 'data',
+                      signal_axes=signal.axes_manager.signal_indices_in_array,
+                      **kwds)
+    # Remove chunks from the kwds since it wouldn't have the same rank as the
+    # dataset and can't be used
+    kwds.pop('chunks', None)
+    dict2zarrgroup(metadata_dict, mapped_par, **kwds)
+    original_par = group.create_group(original_metadata)
+    dict2zarrgroup(signal.original_metadata.as_dictionary(), original_par,
+                   **kwds)
+    learning_results = group.create_group('learning_results')
+    dict2zarrgroup(signal.learning_results.__dict__,
+                  learning_results, **kwds)
+
+    if len(signal.models) and f is not None:
+        model_group = f.require_group('Analysis/models')
+        dict2zarrgroup(signal.models._models.as_dictionary(),
+                      model_group, **kwds)
+        for model in model_group.values():
+            model.attrs['_signal'] = group.name
+
+
+def file_writer(filename, signal, *args, **kwds):
+    """Writes data to hyperspy's zarr format
+
+    Parameters
+    ----------
+    filename: str
+    signal: a BaseSignal instance
+    *args, optional
+    **kwds, optional
+    """
+    if "compressor" not in kwds:
+        from numcodecs import Blosc
+        kwds["compressor"] = Blosc(cname='zstd', clevel=1)
+    store = zarr.storage.NestedDirectoryStore(filename,)
```

Comment:
Add a keyword argument to allow the user to specify the storage class to use. Also maybe a dictionary mapping string to commonly use storage, for example, `directory`, `nested_directory`, `zip`?